src/soko_game.rs

```diff
@@ -106,21 +106,21 @@ pub fn handle_event(model: &mut Model) -> io::Result<Option<GameAction>> {
     //     .debug
     //     .push(format!("{:?}", &window.world.board.dim()));
 
-    if event::poll(Duration::from_millis(250))? {
-        if let Event::Key(key) = event::read()? {
-            if key.kind == event::KeyEventKind::Press {
-                return Ok(handle_key(key));
-            }
-        }
-    }
-
     window.debug.push(format!(
         "\n                Steps: {:?}
         Best Solution: X
         \ntbd... ",
         &model.game.history.len()
     ));
 
+    if event::poll(Duration::from_millis(50))? {
+        if let Event::Key(key) = event::read()? {
+            if key.kind == event::KeyEventKind::Press {
+                return Ok(handle_key(key));
+            }
+        }
+    }
```

Comment:
This fixes the text flash that was happening after the total moves change. I also made the polling period faster so the refresh time isn't as noticeable.

Reply:
excellent! good catch.

Reply:
fix observed to work